test/integration-tests/IntegrationTests.MongoDB/MongoDbCollection.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using DotNet.Testcontainers.Containers.Builders;
+using DotNet.Testcontainers.Containers.Modules;
+using DotNet.Testcontainers.Containers.WaitStrategies;
+using IntegrationTests.Helpers;
+using IntegrationTests.Helpers.Enums;
+using Xunit;
+
+namespace IntegrationTests.MongoDB
+{
+    [CollectionDefinition(Name)]
+    public class MongoDbCollection : ICollectionFixture<MongoDbFixture>
+    {
+        public const string Name = nameof(MongoDbCollection);
+    }
+
+    public class MongoDbFixture : IDisposable
+    {
+        private const int MongoDbPort = 27017;
+
+        private TestcontainersContainer _container;
+
+        public MongoDbFixture()
+        {
+            bool launchContainer = ShouldLaunchContainer();
+
+            Port = launchContainer
+                ? TcpPortProvider.GetOpenPort()
+                : MongoDbPort;
+
+            if (launchContainer)
+            {
+                _container = LaunchMongoContainer(Port);
+            }
+        }
+
+        public int Port { get; }
+
+        public void Dispose()
+        {
+            if (_container != null)
+            {
+                ShutDownMongoContainer(_container);
+            }
+        }
+
+        private bool ShouldLaunchContainer()
+        {
+            var environment = EnvironmentHelper.GetIntegrationsEnvironment();
+
+            if (environment == IntegrationsEnvironment.CI)
```

Comment:
do you suspect any other environments?

if not then we could simply check the `CI` env var. so just change the logic if it is running on CI to something like:

```csharp
private bool IsRunningOnCI() {
  string env = Environment.GetEnvironmentVariable("INTEGRATIONS_ENVIRONMENT");
  return !string.IsNullOrEmpty(env);
}
```

github actions sets `CI=true`. See: https://docs.github.com/en/actions/learn-github-actions/environment-variables

I am only proposing b/c I think it will simplify the code.

If you disagree just resolve this comment

Reply:
Great find, I suspected that there is something already, but could not find it again.